api/api.go

```diff
@@ -0,0 +1,31 @@
+package api
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+)
+
+// respond converts its data parameter to JSON and send it as an HTTP response
+func respond(statusCode int, w http.ResponseWriter, data interface{}) {
+	if statusCode == http.StatusNoContent {
+		w.WriteHeader(statusCode)
+		return
+	}
+
+	// Convert the response value to JSON.
+	jsonData, err := json.Marshal(data)
+	if err != nil {
+		log.Println(err)
```

Comment:
Wondering do we need to add any additional context on this error?

Reply:
My reasoning was that if an error happens on the server end - the client gets a basic "internal server error" message with code 500, but the logs should be able to tell what actually happened.

What do you think of that?

Reply:
Does that error on the server end includes the file and line number? If yes, then yeah that's good.

Reply:
Actually it doesn't. Should probably use zerolog.Logger in that case 🤔